File: tps/tp02/Filme.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.nio.Buffer;

class Filme{
    //variaveis
    private String nome;
    private String titulo_original;
    private String data;
    private int duracao;
    private String genero;
    private String idioma;
    private String situacao;
    private float orcamento;
    private String[] palavras_chaves;

    //construtores
    public Filme(){ 
        nome = "";
        titulo_original = "";
        data = "";
        duracao = 0;
        genero = "";
        idioma = "";
        situacao = "";
        orcamento = 0;
    }

    public Filme(String nome, String titulo_original, String data, int duracao, String genero, String idioma, String situacao, float orcamento, String[] palavras_chaves){
        this.nome = nome;
        this.titulo_original = titulo_original;
        this.data = data;
        this.duracao = duracao;
        this.genero = genero;
        this.idioma = idioma;
        this.situacao = situacao;
        this.orcamento = orcamento;
        this.palavras_chaves = palavras_chaves;
    }

    //getters e setters
    public String getName(){
        return nome;
    }
    public void setName(String nome){
        this.nome = nome;
    }

    public String getTitulo(){
        return titulo_original;
    }
    public void setTitulo(String titulo_original){
        this.titulo_original = titulo_original;
    }

    public String getData(){
        return data;
    }
    public void setData(String data){
        this.data = data;
    }

    public int getDuracao(){
        return duracao;
    }
    public void setDuracao(int duracao){
        this.duracao = duracao;
    }

    public String getGenero(){
        return genero;
    }
    public void setGenero(String genero){
        this.genero = genero;
    }

    public String getIdioma(){
        return idioma;
    }
    public void setIdioma(String idioma){
        this.idioma = idioma;
    }

    public String getSituacao(){
        return situacao;
    }
    public void setSituacao(String situacao){
        this.situacao = situacao;
    }

    public float getOrcamento(){
        return orcamento;
    }
    public void setOrcamento(float orcamento){
        this.orcamento = orcamento;
    }

    public String[] getPalavras(){
        return palavras_chaves;
    }
    public void setPalavras(String[] palavras_chaves){
        this.palavras_chaves = palavras_chaves;
    }

    //filme clone
    protected Filme clone(){
        Filme clone = new Filme();

        clone.nome = this.nome;
        clone.titulo_original = this.titulo_original;
        clone.data = this.data;
        clone.duracao = this.duracao;
        clone.genero = this.genero;
        clone.idioma = this.idioma;
        clone.situacao = this.situacao;
        clone.orcamento = this.orcamento;
        clone.palavras_chaves = this.palavras_chaves;

        return clone;
    }

    //imprimir
    private void imprimir(){
        System.out.println(nome + " ");
    }

    //ler o arquivo (filme)
    public void ler(String entrada){
        try{
            BufferedReader html = new BufferedReader(new InputStreamReader(new FileInputStream("/tmp/filmes/" + entrada)));

            //nome
            for(int i = 0; entrada.charAt(i) != '.'; i++){
                this.nome += entrada.charAt(i);
            }

            html.close();
        } catch(Exception erro){
            System.out.println(erro);
        }

    }

    //fim
    public static boolean isFim(String s){
        return(s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    //main
    public static void main(String[] args){
        MyIO.setCharset("UTF-8");
        Filme[] Filme = new Filme[100];
        String[] entrada = new String[10000];
        int num_linha = 0;

        do{
            entrada[num_linha] = MyIO.readLine();
        }while(isFim(entrada[num_linha++]) == false);
        num_linha--;

        for(int i = 0; i < num_linha; i++){
            Filme[i] = new Filme();
            Filme[i].ler(entrada[i]);
            Filme[i].imprimir();
        }
    }
}

/*Date lancamento

SimpleDateFormat sdf = new SimpleDateFormat("dd/mm/aaaa");

lancamento = sdf.parse("15/02/2021");
*/

File: tps/tp02/TP02Q02 - Registro em C/Filme.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <ctype.h>
#define NUMENTRADA 1000
#define TAMLINHA 1000

// variaveis
typedef struct  {
    char nome[100];
    char titulo_original[100];
    char data[100];
    int duracao;
    char genero[100];
    char idioma[100];
    char situacao[100];
    char orcamento[100];
    char palavras_chave[100];
} Filme;


// construtores
Filme inicializaFilme(){
    Filme filme;
    strcpy(filme.nome, "");
    strcpy(filme.titulo_original, "");
    strcpy(filme.data, "");
    filme.duracao = 0;
    strcpy(filme.genero, "");
    strcpy(filme.idioma, "");
    strcpy(filme.situacao, "");
    strcpy(filme.orcamento, "");
    strcpy(filme.palavras_chave, "");

    return filme;
}

Filme inicializa(char nome[], char titulo_original[], char data[], int duracao, char genero[], char idioma[], char situacao[], char orcamento[], char palavras_chave[]){
    Filme filme;
    strcpy(filme.nome, nome);
    strcpy(filme.titulo_original, titulo_original);
    strcpy(filme.data, data);
    filme.duracao = duracao;
    strcpy(filme.genero, genero);
    strcpy(filme.idioma, idioma);
    strcpy(filme.situacao, situacao);
    strcpy(filme.orcamento, orcamento);
    strcpy(filme.palavras_chave, palavras_chave);

    return filme;
}

// getters e setters
void setNome(Filme *filme, char nome[]){
    strcpy(filme->nome, nome);
}
char *getNome(Filme filme){
    char *a;
    strcpy(a, filme.nome);
    return a;
}

void setTituloOriginal(Filme *filme, char titulo_original[]){
    strcpy(filme->titulo_original, titulo_original);
}
char *getTituloOriginal(Filme filme){
    char *a;
    strcpy(a, filme.titulo_original);
    return a;
}

void setData(Filme *filme, char data[]){
    strcpy(filme->data, data);
}
char *getData(Filme filme){
    char *a;
    strcpy(a, filme.data);
    return a;
}

void setDuracao(Filme *filme, int duracao){
    filme->duracao = duracao;
}
char getDuracao(Filme filme){
    return filme.duracao;
}

void setGenero(Filme *filme, char genero[]){
    strcpy(filme->genero, genero);
}
char *getGenero(Filme filme){
    char *a;
    strcpy(a, filme.genero);
    return a;
}

void setIdioma(Filme *filme, char idioma[]){
    strcpy(filme->idioma, idioma);
}
char *getIdioma(Filme filme){
    char *a;
    strcpy(a, filme.idioma);
    return a;
}

void setSituacao(Filme *filme, char situacao[]){
    strcpy(filme->situacao, situacao);
}
char *getSituacao(Filme filme){
    char *a;
    strcpy(a, filme.situacao);
    return a;
}

void setOrcamento(Filme *filme, char orcamento[]){
    strcpy(filme->orcamento, orcamento);
}
char *getOrcamento(Filme filme){
    char *a;
    strcpy(a, filme.orcamento);
    return a;
}

void setPalavrasChave(Filme *filme, char palavras_chave[]){
    strcpy(filme->palavras_chave, palavras_chave);
}
char *getPalavrasChave(Filme filme){
    char *a;
    strcpy(a, filme.palavras_chave);
    return a;
} 

char *limpaTag(char *old){
    char *newLine = (char *)malloc(sizeof(char) * strlen(old));
    int i = 0, j = 0;
    while (i < strlen(old))
    {
        if (old[i] == '<'){
            i++;
            while (old[i] != '>'){
                i++;
            }
                
        }
        else
        {
            newLine[j] = old[i];
            j++;
        }
        i++;
    }
    newLine[j] = '\0';
    return newLine;
}

char *trimString(char *str){
    char *end;

    while(isspace((unsigned char)*str)) str++;

    if(*str == 0)
        return str;

    end = str + strlen(str) - 1;
    while(end > str && isspace((unsigned char)*end)) end--;

    end[1] = '\0';

    return str;
}

void ler(Filme *Filme, char *nome){
    char filePath[200];
	sprintf(filePath, "/tmp/filmes/%s", nome);
    FILE *fp = fopen(filePath, "r"); 

    char buf[3000];
    while(!strstr(buf, "<title>")){
       fgets(buf, 3000, fp);
    }
    strcpy(Filme->nome,limpaTag(buf));
    Filme->nome[strcspn(Filme->nome, "\n")] = ' ';

    fclose(fp);
}

void imprimir(Filme filme){
    printf("%s\n", filme.nome);
}

int isFim(char s[]){
    return (s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

int main()
{
    char entrada[200][200];
    int numEntrada = 0;
    Filme filmes[61];
    
    do{
        scanf(" %[^\n]s", entrada[numEntrada]);
    } while (isFim(entrada[numEntrada++]) == 0);
    numEntrada--;
    
    for (int i = 0; i < numEntrada; i++){
        filmes[i] = inicializaFilme();
        ler(&filmes[i], entrada[i]);
        imprimir(filmes[i]);
    }

    return 0;
}